src/s390_ecc.c

```diff
@@ -1743,16 +1743,22 @@ struct {				\
 #ifdef ICA_FIPS
 			if (fips & ICA_FIPS_MODE) {
 				fc |= 0x80; /* deterministic signature */
-				RAND_bytes(param.P256.rand + off, sizeof(param.P256.rand) - off);
+				do {
+					/* A cc=2 occurs if the random number is not invertible.
+					 * Loop in this case. */
+					RAND_bytes(param.P256.rand + off, sizeof(param.P256.rand) - off);
+					rc = s390_kdsa(fc, param.buff, NULL, 0);
+				} while (rc);
```

Comment:
I guess we should only loop  for KDSA CC=2 here. If we also loop for other error cases (e.g. invalid private key) it will loop forever.

Reply:
We might also add a repeat counter, and stop repeating after a certain number of repeats. Just in case the random number generator is faulty and always generates the same non-invertible nonce.

Reply:
Yes, a counter definitely makes sense.

Reply:
this could then also cover the 'invalid private key' error case. If it ever happens that a wrong key gets that far into this routine, then it would only loop until the counter hits and fail that way. 

Still, we also could adjust the s390_kdsa function to return the CC to the caller, as suggested in a private email to you.

Reply:
counter is done, but the cc=2 handling is still in discussion